### PR TITLE
feat(route): support category and fix time (twreporter)

### PR DESCRIPTION
**Pull request by emdoe**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/twreporter/category/world
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/twreporter/category.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+async function handler(ctx) {
+    const baseURL = 'https://www.twreporter.org';
+    const url = baseURL + `/categories/${ctx.req.param('category')}`;
+    const res = await got(url);
+    const $ = load(res.data);
+
+    const regexp = /^window\.__REDUX_STATE__=(.*);$/gm;
+    const raw = $('script[charset="UTF-8"]').html().replaceAll(regexp, '$1');
```
Comment (TonyRL) [inline lib/routes/twreporter/category.ts:39]:
```suggestion
    const raw = $('script[charset="UTF-8"]').text().replaceAll(regexp, '$1');
```

File: lib/routes/twreporter/category.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    const ids = list.entities.posts.allIds;
+    const links = [];
+    const titles = [];
+    const category = list.entities.posts.byId[ids[0]].category_set[0].category.name;
+
+    for (const id of ids) {
+        links.push(baseURL + '/a/' + list.entities.posts.byId[id].slug);
+        titles.push(list.entities.posts.byId[id].title);
+    }
```
Comment (TonyRL) [inline lib/routes/twreporter/category.ts:50]:
```suggestion
    const posts = JSON.parse(raw).entities.posts;

    const list = posts.allIds.map((id) => ({
        link: baseURL + '/a/' + posts.byId[id].slug,
        title: posts.byId[id].title,
    }));
    const category = posts.byId[posts.allIds[0]].category_set[0].category.name;
```
Please use map to form an array of objects instead pushing to 2 arrays.

File: lib/routes/twreporter/category.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    const out = await Promise.all(
+        links.map((item, index) => {
+            const title = titles[index];
+            return cache.tryGet(item, async () => {
+                const single = await fetch(item);
+                single.title = title;
+                return single;
+            });
+        })
```
Comment (TonyRL) [inline lib/routes/twreporter/category.ts:60]:
```suggestion
        list.map((item) =>
            cache.tryGet(item.link, async () => {
                const single = await fetch(item.link);
                single.title = item.title;
                return single;
            })
        )
```

**Comment by emdoe** [comment]
Thanks a lot!

**Pull request merged**